types/dom-chromium-ai/index.d.ts

```diff
@@ -0,0 +1,253 @@
+interface WindowOrWorkerGlobalScope {
+    readonly ai: AI;
+}
+
+interface AI {
+    readonly assistant: AIAssistantFactory;
+    readonly summarizer: AISummarizerFactory;
+    readonly writer: AIWriterFactory;
+    readonly rewriter: AIRewriterFactory;
+}
+
+interface AICreateMonitor {
+    ondownloadprogress: (event: DownloadProgressEvent) => void;
+
+    addEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: AddEventListenerOptions | boolean,
+    ): void;
+    removeEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: EventListenerOptions | boolean,
+    ): void;
+}
+
+interface DownloadProgressEvent extends Event {
+    readonly loaded: number;
+    readonly total: number;
+}
+
+type AICreateMonitorCallback = (monitor: AICreateMonitor) => void;
+
+type AICapabilityAvailability = "readily" | "after-download" | "no";
+
+interface AIReadableStream extends ReadableStream<string> {
+    [Symbol.asyncIterator](): AsyncIterableIterator<string>;
+}
+
+// Assistant
+// https://github.com/explainers-by-googlers/prompt-api/#full-api-surface-in-web-idl
+
+interface AIAssistantFactory {
+    create(options?: AIAssistantCreateOptions): Promise<AIAssistant>;
+    capabilities(): Promise<AIAssistantCapabilities>;
+}
+
+interface AIAssistantCreateOptions {
+    signal?: AbortSignal;
+    monitor?: AICreateMonitorCallback;
+
+    systemPrompt?: string;
+    initialPrompts?: AIAssistantPrompt[];
+    topK?: number;
+    temperature?: number;
+}
+
+type AIAssistantPromptRole = "system" | "user" | "assistant";
+
+interface AIAssistantPrompt {
+    role?: AIAssistantPromptRole;
+    content?: string;
+}
+
+interface AIAssistant {
+    prompt(input: string, options?: AIAssistantPromptOptions): Promise<string>;
+    promptStreaming(input: string, options?: AIAssistantPromptOptions): AIReadableStream;
```

Comment:
The result here is a <a href="https://github.com/explainers-by-googlers/prompt-api/#:~:text=optional%20AIAssistantPromptOptions%20options%20%3D%20%7B%7D)%3B-,readablestream,-promptStreaming(DOMString%20input">regular <code>ReadableStream</code></a>.

Reply:
This applies here and everywhere else.

Reply:
FWIW I used `AIReadableStream` here in the past because I got TypeScript errors for iterating on the regular `ReadableStream`, so I added:

```
interface AIReadableStream {
		[Symbol.asyncIterator](): AsyncIterableIterator< string >;
	}
```

Seems like that's somehow not the default in TypeScript? Maybe because Safari doesn't support it? See https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream

Soo.. I think adding `Symbol.asyncIterator` might be needed (unless tests prove this wrong)

Reply:
Done! TypeScript 5.5 has a compatible [ReadableStream](https://github.com/microsoft/TypeScript/blob/a709f9899c2a544b6de65a0f2623ecbbe1394eab/src/lib/dom.asynciterable.generated.d.ts#L20) implementation in `dom.asynciterable`.